harness/determined/_core/_distributed.py

```diff
@@ -355,3 +405,10 @@ def __init__(self) -> None:
             cross_rank=0,
             cross_size=1,
         )
+
+
+def _get_default_port_offset() -> int:
```

Comment:
is this helper function necessary? the port offset is defaulted to 0 in init. also will this ever be called outside of a trial task type?

Reply:
My reasoning was: there are dtrain scenarios where we need to port offsets to ensure multiple sub-entire-node dtrain jobs can coexist on a node.  I don't think there's any other reason to set that port offset.  When you create the `DistributedContext()` manually (the lowest-level API) you can set that value, but that's the sort of detail I'd rather not expose in the helper functions like `DistributedContext.from_horovod()` or `DistributedContext.from_deepspeed()`.  But I'd like those helper functions to do the right thing by default when they are used in sub-entire-node dtrain jobs, so picking the right port_offset somehow is necessary.

And since you'll be adding a `DistributedContext.from_torch_distributed()` soon, having 3 copies of that logic seems like its worth having a helper function.  It doesn't get exposed outside of this file though, which I like.

Reply:
Perhaps "default" is not a good name for it.  I'm open to renaming it.  "_get_training_port_offset()" maybe?

Reply:
i think `_get_training_port_offset` would make more sense, implying that different implementations can set different offsets